From a58dfdbc9ca27989bd2bcfed141051196354881f Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Fri, 22 Mar 2024 13:07:00 +0000
Subject: [PATCH] trivial: Include the type in the UEFI capsule device summary
 (#6958)

* trivial: Include the type in the UEFI capsule device summary

This means we can spot CoD just from a fwupdmgr get-devices.

Co-authored-by: Mario Limonciello <mario.limonciello@amd.com>
(cherry picked from commit 1b8f1b5dcf448090214e00ee7c76e5c36b2fef9a)
---
 plugins/uefi-capsule/fu-uefi-cod-device.c   | 2 ++
 plugins/uefi-capsule/fu-uefi-device.c       | 1 -
 plugins/uefi-capsule/fu-uefi-grub-device.c  | 2 ++
 plugins/uefi-capsule/fu-uefi-nvram-device.c | 2 ++
 4 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/plugins/uefi-capsule/fu-uefi-cod-device.c b/plugins/uefi-capsule/fu-uefi-cod-device.c
index 93d39945e..5b823365f 100644
--- a/plugins/uefi-capsule/fu-uefi-cod-device.c
+++ b/plugins/uefi-capsule/fu-uefi-cod-device.c
@@ -258,6 +258,8 @@ fu_uefi_cod_device_report_metadata_pre(FuDevice *device, GHashTable *metadata)
 static void
 fu_uefi_cod_device_init(FuUefiCodDevice *self)
 {
+	fu_device_set_summary(FU_DEVICE(self),
+			      "UEFI System Resource Table device (Updated via caspule-on-disk)");
 }
 
 static void
diff --git a/plugins/uefi-capsule/fu-uefi-device.c b/plugins/uefi-capsule/fu-uefi-device.c
index 0a232970a..a0ae4af5e 100644
--- a/plugins/uefi-capsule/fu-uefi-device.c
+++ b/plugins/uefi-capsule/fu-uefi-device.c
@@ -685,7 +685,6 @@ fu_uefi_device_set_progress(FuDevice *self, FuProgress *progress)
 static void
 fu_uefi_device_init(FuUefiDevice *self)
 {
-	fu_device_set_summary(FU_DEVICE(self), "UEFI ESRT device");
 	fu_device_add_protocol(FU_DEVICE(self), "org.uefi.capsule");
 	fu_device_add_internal_flag(FU_DEVICE(self), FU_DEVICE_INTERNAL_FLAG_MD_SET_SIGNED);
 	fu_device_add_internal_flag(FU_DEVICE(self), FU_DEVICE_INTERNAL_FLAG_MD_SET_FLAGS);
diff --git a/plugins/uefi-capsule/fu-uefi-grub-device.c b/plugins/uefi-capsule/fu-uefi-grub-device.c
index e4d0eb3e3..51e413aa5 100644
--- a/plugins/uefi-capsule/fu-uefi-grub-device.c
+++ b/plugins/uefi-capsule/fu-uefi-grub-device.c
@@ -198,6 +198,8 @@ fu_uefi_grub_device_report_metadata_pre(FuDevice *device, GHashTable *metadata)
 static void
 fu_uefi_grub_device_init(FuUefiGrubDevice *self)
 {
+	fu_device_set_summary(FU_DEVICE(self),
+			      "UEFI System Resource Table device (updated via grub)");
 }
 
 static void
diff --git a/plugins/uefi-capsule/fu-uefi-nvram-device.c b/plugins/uefi-capsule/fu-uefi-nvram-device.c
index 107f24066..0acd770b4 100644
--- a/plugins/uefi-capsule/fu-uefi-nvram-device.c
+++ b/plugins/uefi-capsule/fu-uefi-nvram-device.c
@@ -133,6 +133,8 @@ fu_uefi_nvram_device_report_metadata_pre(FuDevice *device, GHashTable *metadata)
 static void
 fu_uefi_nvram_device_init(FuUefiNvramDevice *self)
 {
+	fu_device_set_summary(FU_DEVICE(self),
+			      "UEFI System Resource Table device (updated via NVRAM)");
 }
 
 static void
-- 
2.34.1

